src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -278,6 +284,87 @@ internal static object GetClaimValueUsingValueType(Claim claim)
             return claim.Value;
         }
 
+        /// <summary>
+        /// Gets ValueType of the claim from it's Value.
+        /// </summary>
+        /// <param name="value">Represents value of a claim.</param>
+        /// <returns>String representing claim's ValueType.</returns>
+        internal static string GetClaimValueTypeFromValue(object value)
+        {
+            if (value.GetType().Name == typeof(String).Name && value.ToString() == string.Empty)
+                return JsonClaimValueTypes.JsonNull;
+
+            string claimValueType = TokenUtilities.GetClaimValueTypeFromValue(value);
+
+            if (!String.IsNullOrEmpty(claimValueType))
+                return claimValueType;
+
+            if (value.GetType().Name == typeof(JObject).Name)
+            {
+                return JsonClaimValueTypes.Json;
+            }
+            if (value.GetType().Name == typeof(JArray).Name)
+            {
+                return JsonClaimValueTypes.JsonArray;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Creats claims from dictionary.
+        /// </summary>
+        /// <param name="claimsCollection">Represents claims stored in dictionary.</param>
+        /// <returns>claims</returns>
+        internal static IEnumerable<Claim> CreateClaimsFromDictionary(IDictionary<string, object> claimsCollection)
+        {
+            List<Claim> claims = new List<Claim>();
+            object value;
+            foreach (string claimtype in claimsCollection.Keys)
+            {
+                claimsCollection.TryGetValue(claimtype, out value);
+                if (value != null)
```

Comment:
It is conventional to use the bool returned by `TryGetValue` to check for existence

https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.idictionary-2.trygetvalue?view=netcore-3.1

Reply:
Would it be simpler to just iterate through the 'key, value' pairs instead of calling TryGetValue? 

`foreach(var entry in claimsCollection)`

Then you can access `entry.Key` and `entry.Value`.

Reply:
I find it repeating to use entry.Key and entry.Value every time I need them. I can define a variable for each of them and use , but this necessarily is doing the same.